lib/elixir/lib/calendar/iso.ex

```diff
@@ -220,26 +220,43 @@ defmodule Calendar.ISO do
   @doc """
   Converts the given time into a string.
   """
-  def time_to_string(hour, minute, second, {_, 0}) do
-    time_to_string(hour, minute, second)
+  def time_to_string(hour, minute, second, microsecond, format \\ :extended)
+  def time_to_string(hour, minute, second, {_, 0}, format) do
+    case format do
+      :extended -> extended_time_to_string(hour, minute, second)
```

Comment:
I think `time_to_string_extended` and `time_to_string_basic` would be better naming for these functions. It reads better that way (since it is not about extended or basic time but about output string), also keeps them grouped by common prefix.